sgt-back-end/index.js

```diff
@@ -0,0 +1,262 @@
+        res.status(500).json({
+          error: 'An unexpected error occurred.'
+        });
+      });
+  }
+});
+
+app.delete('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+
+  const sql = `
+     delete from "grades"
+     where "gradeId" = $1
+     returning *
+  `;
+  const params = [gradeId];
+
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows;
+      if (grade && grade.length > 0) {
+        res.sendStatus(204);
+      } else {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  const newMessage = req.body;
+
+  const newName = newMessage.name;
+  const newCourse = newMessage.course;
+  const newScore = parseInt(newMessage.score);
+
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+
+  } else if (!newScore) {
+    res.status(400).json({
+      error: '"score" is required'
+    });
+
+  } else if (!Number.isInteger(newScore) || newScore < 0 || newScore > 100) {
+    res.status(400).json({
+      error: '"score" must be between 0 and 100'
+    });
+
+  } else if (!newName) {
+    res.status(400).json({
+      error: '"name" is required'
+    });
+
+  } else if (!newCourse) {
+    res.status(400).json({
+      error: '"course" is required'
+    });
+
+  } else {
+    const sql = `
+     delete from "grades"
+     where "gradeId" = $1
+     returning *
+  `;
+    const params = [gradeId];
+    db.query(sql, params)
+      .then(result => {
+        const grade = result.rows;
+        if (grade && grade.length > 0) {
+          const sql2 = `insert into "grades"
+          ("name", "course", "score")
+          values ($1, $2, $3)
+          returning *
+          `;
+          const params2 = [newName, newCourse, newScore];
+          db.query(sql2, params2)
+            .then(result => {
+              const grade = result.rows[0];
+              res.json(grade);
+            })
+            .catch(err => {
+              console.error(err);
+              res.status(500).json({
+                error: 'An unexpected error occurred.'
+              });
+            });
+        } else {
+          res.status(404).json({
+            error: `Cannot find grade with "gradeId" ${gradeId}`
+          });
+        }
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({
+          error: 'An unexpected error occurred.'
+        });
+      });
```

Comment:
Instead of deleting then creating a new row we want to update the existing row using an `UPDATE` query:

```SQL
update "grades"
  set 
      "name" = $1,
      "course" = $2,
      "score" = $3
 where 
   "gradeId" = $4;
 ```